Source/WebCore/platform/network/BlobRegistryImpl.cpp

```diff
@@ -120,6 +120,42 @@ void BlobRegistryImpl::registerFileBlobURL(const URL& url, Ref<BlobDataFileRefer
     addBlobData(url.string(), WTFMove(blobData));
 }
 
+static FileSystem::MappedFileData storeInMappedFileData(const String& path, const uint8_t* data, size_t size)
+{
+    auto mappedFileData = FileSystem::createMappedFileData(path, size);
+    if (!mappedFileData)
+        return { };
+    FileSystem::deleteFile(path);
+
+    memcpy(const_cast<void*>(mappedFileData.data()), data, size);
+
+    FileSystem::finalizeMappedFileData(mappedFileData, size);
+    return mappedFileData;
+}
+
+Ref<DataSegment> BlobRegistryImpl::createDataSegment(Vector<uint8_t>&& movedData, BlobData& blobData)
+{
+    ASSERT(isMainThread());
+
+    auto data = DataSegment::create(WTFMove(movedData));
+    if (m_fileDirectory.isEmpty())
+        return data;
+
+    static uint64_t blobMappingFileCounter;
+    static NeverDestroyed<Ref<WorkQueue>> workQueue(WorkQueue::create("BlobRegistryImpl Data Queue"));
+    auto filePath = FileSystem::pathByAppendingComponent(m_fileDirectory, makeString("mapping-file-", ++blobMappingFileCounter, ".blob"));
+    workQueue.get()->dispatch([blobData = Ref { blobData }, data, filePath = WTFMove(filePath).isolatedCopy()]() mutable {
+        auto mappedFileData = storeInMappedFileData(filePath, data->data(), data->size());
+        if (!mappedFileData)
+            return;
+        ASSERT(mappedFileData.size() == data->size());
+        callOnMainThread([blobData = WTFMove(blobData), data = WTFMove(data), newData = DataSegment::create(WTFMove(mappedFileData))]() mutable {
+            blobData->replaceData(data.get(), WTFMove(newData));
+        });
```

Comment:
Is it possible to write blob data to our mapped file eagerly, rather than all at once at the end? Writing at the end will solve jetsam in the case of many small or medium sized blobs, but not in the case of one large blob.

Reply:
We are already storing blobs in network process as a vector of BlobDataItem, we do not concatenate the Blob parts in one big chunk of data before storage.

Reply:
Would be nice to improve on this in a follow-up. @achristensen07 has shown me examples in the past where one big blob would take down the Network process. I think we can use this file-backed technique to fix them.

Reply:
If we create a big blob from multiple small blobs, the memory impact should be limited.
If several pages create big blobs, this patch should help.
If one page creates a big blob that takes down the network process, this patch will not help.